x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityBwcRestIT.java

```diff
@@ -0,0 +1,245 @@
+    }
+
+    @ClassRule
+    // Use a RuleChain to ensure that fulfilling cluster is started before query cluster
+    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
+
+    public void testBwcWithLegacyCrossClusterSearch() throws Exception {
+        final boolean useProxyMode = randomBoolean();
+        // Update remote cluster settings on QC.
+        setupQueryClusterRemoteClusters(useProxyMode);
+        // Ensure remote cluster is connected
+        ensureRemoteFulfillingClusterIsConnected(useProxyMode);
+
+        // Fulfilling cluster
+        {
+            // Index some documents, so we can attempt to search them from the querying cluster
+            final Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                { "index": { "_index": "index1" } }
+                { "foo": "bar" }
+                { "index": { "_index": "index2" } }
+                { "bar": "foo" }
+                { "index": { "_index": "prefixed_index" } }
+                { "baz": "fee" }\n"""));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+        }
+
+        // Query cluster
+        {
+            // Index some documents, to use them in a mixed-cluster search
+            final var indexDocRequest = new Request("POST", "/local_index/_doc?refresh=true");
+            indexDocRequest.setJsonEntity("{\"local_foo\": \"local_bar\"}");
+            assertOK(client().performRequest(indexDocRequest));
+
+            // Create user role with privileges for remote and local indices
+            final var putRoleRequest = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+            putRoleRequest.setJsonEntity("""
+                {
+                  "cluster": ["manage_own_api_key"],
+                  "indices": [
+                    {
+                      "names": ["local_index", "index1", "not_found_index"],
+                      "privileges": ["read", "read_cross_cluster"]
+                    }
+                  ],
+                  "remote_indices": [
+                    {
+                      "names": ["index1", "not_found_index", "prefixed_index"],
+                      "privileges": ["read", "read_cross_cluster"],
+                      "clusters": ["my_remote_cluster"]
+                    }
+                  ]
+                }""");
+            assertOK(adminClient().performRequest(putRoleRequest));
+            // We need to define the same role on QC and FC in order for CCS to work.
+            final var putRoleRequestFulfilling = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+            putRoleRequestFulfilling.setJsonEntity("""
+                {
+                  "cluster": ["manage_own_api_key"],
+                  "indices": [
+                    {
+                      "names": ["local_index", "index1", "not_found_index"],
```

Comment:
Nit: I think this should mirror the `remote_indices` definition for the role: that way a legacy FC and new FC should have the same role definitions for a QC-side user.